### DISCUSSION TITLE
Universal X86 Tuning Utility V3 - Adaptive Mode Rewrite

### DISCUSSION THREAD

**Discussion by JamesCJ60** [JamesCJ60/Universal-x86-Tuning-Utility#254] [Announcements]
Hi all, 
I just wanted to share some developments regarding the new adaptive mode with UXTU V3, which will hopefully be released within the next year!

The new adaptive mode now has its own machine-learning algorithm that trains itself over time on a per-application basis, allowing it to make intelligent adjustments specifically for your devices. These adjustments are made based on your current/average frame rate within games and the current temperature at which your laptop/PC is running. Using a reward system, past data is used to determine the best adjustments at a given time. It can also work off just temperatures like the old algorithm as a fallback if FPS data is unavailable.

Currently, this is up and running within a test environment on a Framework Laptop 16 with a Ryzen 9 7940HS and RX 7700S utilising SmartShift. I have included screenshots of this working in some games I tested below. I will provide a video demo in the not-so-distant future.

**The Last of Us Part 1**
![image](https://github.com/user-attachments/assets/f131e810-9ade-454d-8323-ad5612407e03)

**Spider-Man Remastered**
![image](https://github.com/user-attachments/assets/99472698-6e67-4a92-91a4-f937fd62c1af)

**Dead Space**
![image](https://github.com/user-attachments/assets/4b176da9-1dcc-417e-a434-7c968cc6790c)

**Fallout 4**
![image](https://github.com/user-attachments/assets/50ae4313-467d-4c68-86a3-33e2d56766d0)

This still needs some work, but it is heading in the right direction. The following steps will aim to enhance the TDP control machine-learning algorithm further and create a machine-learning algorithm to handle iGPU clock adjustments for AMD and Intel systems on top of an undervolting system.

I want to thank everyone who has supported this project in many different ways over the years. This would not have been possible without you!

**Comment by Erruar** [comment]
Hello, I'm **Serzhik Sakurajima**  
*Developer of Saku Overclock, a software solution for overclocking laptops with AMD Ryzen processors.*

### After reading this news, I have several important questions. 
*I'd be glad to see your explanations on them.*

---

**1. Support and Compatibility with AMD Ryzen Processors**  
- Will the new adaptive system work across all AMD Ryzen processors?  
- If not, what specific limitations or requirements might affect compatibility?

---

**2. Performance and Resource Usage**  
- How much CPU and/or GPU resources will the new system require, considering its AI-based algorithm and FPS monitoring?  
- Could this system significantly impact performance during intensive workloads or gaming sessions?

---

**3. Safety and Adjustment Limits**  
- What safety mechanisms are in place in the adaptive system to prevent overheating or potential damage to the CPU?  
- Are there specific boundaries for adjustments, and can they be customized?  
- How safe is it to apply these methods on high-performance processors, which already have high TDP and frequency levels by default?

---

**4. Technical Implementation and Architecture**  
- What technologies are being used to implement dynamic control of parameters? For example, will LLM or other algorithms be employed for prediction and adaptation?  
- Could you provide an overview of the architecture or working scheme for this mechanism?  
- How frequently are temperature and FPS data updated to ensure accurate adjustments?

---

**5. Value and Relevance of the Innovation**  
- Why do you consider this change to be a significant step forward?  
- How specifically could this innovation improve system efficiency, especially compared to existing TDP and frequency adjustment algorithms?

---

**6. Additional Questions**  
- Will there be an option to test the solution on other AMD Ryzen processors and see its effectiveness in real-world scenarios?  
- Are there any plans to extend this system to other manufacturers, such as Intel processors?
